Update quickstart for extension v2 (#1327)
databricks/databricks-vscode: packages/databricks-vscode/DATABRICKS.quickstart.md

> # Databricks Extension for Visual Studio Code

Reviewer: ```suggestion
# Databricks extension for Visual Studio Code
```

> The Databricks extension for VS Code allows you to develop for the Databricks Lakehouse platform from VS Code.

Reviewer: ```suggestion
The Databricks extension for Visual Studio Code allows you to develop for the Databricks Data Intelligence Platform from Visual Studio Code.
```
This should really match the first line in the docs, which is currently "The Databricks extension for Visual Studio Code enables you to connect to your remote Databricks workspaces from Visual Studio Code." But it could be I should change the docs instead. Thoughts?

> # ⚠️ Note

Reviewer 2: Let's remove the "Note" too?

# Databricks Extension for Visual Studio Code

> The Databricks extension for VS Code allows you to develop for the Databricks Lakehouse platform from VS Code.

Reviewer: Yes let's use the official product name (see suggestion).

> 

Reviewer: How about "See the [Databricks documentation](https://docs.databricks.com/dev-tools/vscode-ext.html) for complete usage information."

(Or

# Features

> -   Create [Databricks Asset Bundle](https://docs.databricks.com/en/dev-tools/bundles/index.html)

Reviewer 2: Plural "Bundles" sounds better here, but maybe it's just me

Reviewer: Yes, we need to always use "Databricks Asset Bundles" (plural), as in this case, or just "bundle" if singular is required (e.g., bundle configuration file, bundle root, etc.).

> -   Create [Databricks Asset Bundle](https://docs.databricks.com/en/dev-tools/bundles/index.html)

Reviewer: For this list, again, maybe we should match the docs more closely? It's pretty close, but: 

- Define, deploy, and run Databricks Asset Bundles to apply CI/CD patterns to your Databricks jobs, Delta Live Tables pipelines, and MLOps Stacks.
- Run local Python code files on Databricks clusters.
- Run notebooks and local Python code files as Databricks jobs. 
- Set up and configure your debugging environment and Databricks Connect using a simple checklist that triggers selection dialogs.
- Debug notebooks cell by cell with Databricks Connect.
- Synchronize local code with code in your Databricks workspace.

-   Manage, run and deploy Databricks assets (jobs, DLT Pipelines)
-   Synchronize code to a Databricks workspace
-   Run Python files on a Databricks cluster
-   Run notebooks and Python files as Workflows
-   Debug Spark code running on a Databricks cluster using Databricks Connect
-   Run notebooks cell by cell using Databricks Connect

## <a id="toc"></a>Table of Contents

-   [Getting Started](#setup-steps)

>     -   [Create a Databricks Project](#create-databricks-project)

Reviewer: ```suggestion
    -   [Create a Databricks project](#create-databricks-project)
```

>     -   [Running Code](#running-code)

Reviewer: Maybe instead "Run Python code" (or file)?

        -   [Running PySpark Code](#running-pyspark-code)
        -   [Running PySpark Code and Notebooks as Workflows](#running-code-as-workflows)

>     -   [Deploying Databricks Asset Bundle](#dabs)

Reviewer 2: Similar to above, maybe let's make it plural here?

Reviewer: Yes. (See above.) Since DABs is already spelled out above, it would also be OK to just use "bundle" here.

        -   [What is a Databricks Asset Bundle?](#what-is-dab)
        -   [Deploying a Databricks Asset Bundle](#deploy-dab)
        -   [Run a Job or Pipeline](#deploy-run-job-pipeline)
-   [Changes from v1](#changes-from-v1)
    -   [Migrate a project from Databricks Extension v1 to v2](#migrate-from-v1)
    -   [What is databricks.yml?](#what-is-databricksyaml)
    -   [No environment variables in terminals](#no-env-vars)

---

# <a id="setup-steps"></a>Getting Started

## <a id="create-databricks-project"></a>Create a Databricks Project

1. Open the Databricks panel by clicking on the Databricks icon on the left
2. Click on the "Create a new Databricks Project" button.

> 3. Follow the wizard to select an existing Databricks Profile or create a new one.

Reviewer: ```suggestion
3. Follow the selection dialogs to create a Databricks configuration profile or select an existing one.
```

4. Select a folder to create your project in.

> 5. Follow the wizard to create a new Databricks Project.

Reviewer: ```suggestion
5. Follow the selection dialogs to create a new Databricks project.
```

6. Select the newly created project to open it, using the selector that appears.
7. VS Code will reopen with the new project loaded, and the extension will automatically login using the selected profile.

![create-databricks-project](./images/dabs_vsc.gif)

If your folder has multiple [Databricks Asset Bundles](#dabs), you can select which one to use by clicking "Open Existing Databricks Project" button and selecting the desired project.

## <a id="select-cluster"></a>Select a cluster

> The extension uses an interactive cluster to run code. To select an interactive cluster

Reviewer: ```suggestion
The extension uses an interactive cluster to run code. To select an interactive cluster:
```

> 1. Open the Databricks panel by clicking on the Databricks icon on the left

Reviewer: ```suggestion
1. Open the Databricks extension panel by clicking on the Databricks icon on the left sidebar.
```

The extension uses an interactive cluster to run code. To select an interactive cluster

1. Open the Databricks panel by clicking on the Databricks icon on the left
2. Click on the "Select Cluster" button.
    - If you wish to change the selected cluster, click on the "Configure Cluster" gear icon, next to the name of the selected cluster.

## <a id="running-code"></a>Running Code

Once you have your project configured you can sync your local code to the repo and run it on a cluster. You can use the https://github.com/databricks/ide-best-practices repository as an example.
Once you have your project configured you can deploy your local code to the selected Databricks workspace and run it on a cluster.

### <a id="running-pyspark-code"></a>Running PySpark code

1. Create python file
2. Add PySpark code to the python file.
3. Click the "Run" icon in the tab bar and select "Upload and Run File on Databricks"

This will start the code synchronization and run the active python file on the configured cluster. The result is printed in the "debug" output panel.
3. Click the "Databricks Run" icon in the tab bar and select "Upload and Run File on Databricks"

> ![run](./images/run.gif)

Reviewer 2: Let's add a screen recording showcasing the flow?

Reviewer 3: Would be great it we could add some more visuals to it (if possible)

> ## <a id="dabs"></a>Deploying Databricks Asset Bundle

Reviewer 2: Similar to the other comment - let's make it plural or add an "a"

> 1. In the Databricks panel, look for the "Bundle Resource Explorer" view.

Reviewer: ```suggestion
1. In the Databricks panel, find the "Bundle Resource Explorer" view.
```

2. Click on the "Deploy" button.

> 3. You can monitor the deployment status in the log output window.

Reviewer 2: Let's add a gif for this flow too? And for the "Run a Job"

### <a id="deploy-run-job-pipeline"></a>Run a Job or Pipeline

> You can run a job or a pipeline managed by Databricks Asset Bundles, from the "Bundle Resource Explorer" view.

Reviewer: ```suggestion
You can run a job or a pipeline managed by Databricks Asset Bundles, from the "Bundle Resource Explorer" pane.
```

> 1. In the Databricks panel, look for the "Bundle Resource Explorer" view.

Reviewer: ```suggestion
1. In the Databricks panel, find the "Bundle Resource Explorer" pane.
```

> 2. Hover on the line with the job or pipeline you want to run.

Reviewer: ```suggestion
2. Hover over the line with the job or pipeline that you want to run.
```

3. Click on the "Run" button.

> This will deploy the Databricks Asset Bundle and run the job or pipeline. You can monitor the run progress in the output terminal window. You can also open the run, job or pipeline in workspace by clicking on the "Open link externally" button.

Reviewer: ```suggestion
This deploys the bundle and runs the job or pipeline. You can monitor the run progress in the output terminal window. You can also open the run, job or pipeline in workspace by clicking on the "Open link externally" button.
```

> #### Use interactive cluster for running jobs

Reviewer: ```suggestion
#### Use an interactive cluster for running jobs
```

> By default, a job is run using a jobs cluster. You can change this behaviour and use the interactive cluster selected previously ([Select a cluster](#select-cluster)) to run the job.

Reviewer: I think it's fine to leave it, but just a callout that American English is "behavior".

> 1. In the Databricks panel, look for the "Configuration" view.

Reviewer: ```suggestion
1. In the Databricks panel, find the "Configuration" pane.
```

2. Check the "Override Jobs cluster in bundle" checkbox.

> # <a id="changes-from-v1"></a> Key behaviour changes for users of Databricks Extension v1

Reviewer: ```suggestion
# <a id="changes-from-v1"></a> Key behaviour changes for users of Databricks extension v1
```

## <a id="migrate-from-v1"></a>Migrate a project from Databricks Extension v1 to v2

> If you are using Databricks Extension v1, will automatically migrate your project to v2 when you open it in v2. The migration process will create a new [`databricks.yml`](#what-is-databricksyaml) file in the root of your project and move the configurations from the old `.databricks/project.json` to the new `databricks.yml` file.

Reviewer 2: ```suggestion
If you are using Databricks Extension v1, we will automatically migrate your project to an asset bundle when you open it in v2. The migration process will create a new [`databricks.yml`](#what-is-databricksyaml) file in the root of your project and move the configurations from the old `.databricks/project.json` to the new `databricks.yml` file.
```

> If you are using Databricks Extension v1, will automatically migrate your project to v2 when you open it in v2. The migration process will create a new [`databricks.yml`](#what-is-databricksyaml) file in the root of your project and move the configurations from the old `.databricks/project.json` to the new `databricks.yml` file.

Reviewer: I've been trying to avoid references to "v1" and "v2" and use "earlier versions".

> If you are using Databricks Extension v1, will automatically migrate your project to v2 when you open it in v2. The migration process will create a new [`databricks.yml`](#what-is-databricksyaml) file in the root of your project and move the configurations from the old `.databricks/project.json` to the new `databricks.yml` file.

Reviewer: ```suggestion
If you are using Databricks Extension v1, your project will automatically be migrated to v2 when you open it in v2. The migration process will create a new [`databricks.yml`](#what-is-databricksyaml) file in the root of your project and move the configurations from the old `.databricks/project.json` to the new `databricks.yml` file.
```

> > **Note**: This means that you will start seeing a `databricks.yml` file in your project root directory and in your VCS change logs. We recomment comitting this file to your VCS.

Reviewer: What is VCS? I assume version control system, but can we spell it out?

(Also typos "We recommend committing this file to your VCS.")

> ## <a id="what-is-databricksyaml"></a>What is databricks.yaml?

Reviewer: ```suggestion
## <a id="what-is-databricksyaml"></a>What is databricks.yml?
```

> A `databricks.yml` file is a configuration file that describes a Databricks Asset Bundle. It contains the configurations such as the workspace host and description of resources such as jobs and piplines. For more information on `databricks.yml`, refer to the [full docs](https://docs.databricks.com/en/dev-tools/bundles/work-tasks.html#create-a-bundle-manually).

Reviewer: ```suggestion
A `databricks.yml` file is a configuration file that describes a bundle. It contains the configuration such as the workspace host and definitions of resources such as jobs and pipelines. For more information on `databricks.yml`, refer to the [full docs](https://docs.databricks.com/en/dev-tools/bundles/index.html).
```

## <a id="no-env-vars"></a>No environment variables in terminals

> We don't support environment variables in terminals anymore. If you were using environment variables in v1, you will need to manually load the `.databricks/.databricks.env` file in your terminal before running any commands.

Reviewer: ```suggestion
Environment variables in terminals is no longer supported. If you were using environment variables in v1, you will need to manually load the `.databricks/.databricks.env` file in your terminal before running any commands.
```

# Features

> -   Create a Databricks Project [Databricks Asset Bundle]()

Reviewer 2: I think it should just be "Create Databricks Asset Bundles" (+ the right doc link)

> -   Manage, run and deploy Databricks assets (jobs, DLT Pipelines)

Reviewer: I wouldn't link to more than the top DABs article honestly. (But we could rephrase that DABs bullet in both the docs and here though.)

1. Open the Databricks panel by clicking on the Databricks icon on the left
2. Click on the "Create a new Databricks Project" button.
3. Follow the wizard to select an existing Databricks Profile or create a new one.
4. Select a folder to create your project in.
5. Follow the wizard to create a new Databricks Project.
6. Select the newly created project to open it, using the selector that appears.
7. VS Code will reopen with the new project loaded, and the extension will automatically login using the selected profile.

![create-databricks-project](./images/dabs_vsc.gif)

If your folder has multiple [Databricks Asset Bundles](#dabs), you can select which one to use by clicking "Open Existing Databricks Project" button and selecting the desired project.

## <a id="select-cluster"></a>Select a cluster

The extension uses an interactive cluster to run code. To select an interactive cluster

1. Open the Databricks panel by clicking on the Databricks icon on the left
2. Click on the "Select Cluster" button.
    - If you wish to change the selected cluster, click on the "Configure Cluster" gear icon, next to the name of the selected cluster.

## <a id="running-code"></a>Running Code

> Once you have your project configured you can deploy your local code to the repo and run it on a cluster.

Reviewer 2: > ... to the repo

to the workspace?
